src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/TypeLoaderExports.cs

```diff
@@ -123,6 +124,40 @@ public static object GenericLookupAndCast(object arg, IntPtr context, IntPtr sig
             return RawCalliHelper.Call<object>(v._result, arg, v._auxResult);
         }
 
+        public static unsafe IntPtr InterfaceLookupForSlot(object obj, IntPtr slot)
+        {
+            // TODO: do something more optimal for monomorphic case
```

Comment:
I assume here you will dereference the slot/cell and if obj.GetMethodTable() matches the last seen MT, just return the last used callee?

Perhaps if the bigger cache lookup and the cell update is in a separate method, the monomorphic check may even be small enough method to be inlineable into the caller. Can these kinds of helpers be inlined by the JIT?

Reply:
> Can these kinds of helper calls be inlined by the JIT?

These helpers look the same to the JIT as e.g. casting helpers. IIRC JIT doesn't know they're in managed code and it could ask for the IL.